app/qml/inputs/MMTextBubblesEditor.qml

```diff
@@ -0,0 +1,168 @@
+ ***************************************************************************/
+
+import QtQuick
+import QtQuick.Controls
+import QtQuick.Controls.Basic
+import "../components"
+
+MMAbstractEditor {
+  id: root
+
+  property var parentValue: parent.value ?? ""
+  property bool parentValueIsNull: parent.valueIsNull ?? false
+  property bool isReadOnly: parent.readOnly ?? false
+
+  readonly property real itemHeight: 36 * __dp
+  readonly property int rows: 3
+  required property ListModel featuresModel
+
+  signal editorValueChanged( var newValue, var isNull )
+  signal openLinkedFeature( var linkedFeature )
+  signal createLinkedFeature( var parentFeature, var relation )
+
+  contentItemHeight: itemHeight * rows + 2 * flow.spacing + 20 * __dp
+
+  Component.onCompleted: root.hideFeaturesIfNeeded()
+  onWidthChanged: root.hideFeaturesIfNeeded()
+
+  content: Rectangle {
+    width: root.width - 2 * root.spacing
+    height: root.contentItemHeight
+    color: __style.whiteColor
+
+    Flow{
+      id: flow
+
+      anchors.fill: parent
+      anchors.margins: 10 * __dp
+      spacing: 8 * __dp
+      clip: true
+
+      Rectangle {
+        width: 40 * __dp
+        height: root.itemHeight
+        radius: 8 * __dp
+        color: __style.lightGreenColor
+
+        MMIcon {
+          anchors.centerIn: parent
+          source: __style.plusIcon
+          useCustomSize: true
+          width: 16 * __dp
+          height: width
+        }
+
+        MouseArea {
+          anchors.fill: parent
+          onClicked: root.createLinkedFeature( root.parent.featurePair, root.parent.associatedRelation )
+        }
+      }
+
+      Repeater {
+        id: repeater
+
+        property var invisibleIds: []
+
+        model: root.featuresModel
+
+        delegate: Rectangle {
+          width: text.contentWidth + 24 * __dp
+          height: root.itemHeight
+          radius: 8 * __dp
+          color: __style.mediumGreenColor
+          visible: {
+            let isVisible = (y < 2 * root.itemHeight || (y < 3 * root.itemHeight && x + width < flow.width - footer.width - flow.spacing))
+            if(!isVisible && !repeater.invisibleIds.includes(model.FeatureId))
+              repeater.invisibleIds.push(model.FeatureId)
+            footer.visible = repeater.invisibleIds.length > 0
+            moreText.text = ("+" + repeater.invisibleIds.length + qsTr(" more"))
+            return isVisible
+          }
+
+          Text {
+            id: text
+
+            anchors.centerIn: parent
+            font: __style.t4
+            text: model.FeatureTitle
+            color: __style.forestColor
+            horizontalAlignment: Text.AlignHCenter
+            verticalAlignment: Text.AlignVCenter
+          }
+
+          MouseArea {
+            anchors.fill: parent
+            onClicked: root.openLinkedFeature( model.FeaturePair )
+          }
+        }
+      }
+
+      Rectangle {
+        id: footer
+
```

Comment:
Aren't you missing something like:
```qml
    visible: repeater.invisibleIds > 0
```
here? To hide this item if there is enough space to show all the items

Reply:
no, due to an odd behavior of the invisibleIds's length